src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/DBFilterBase.java

```diff
@@ -288,9 +288,81 @@ boolean canAddField() {
       }
     }
   }
+
+  /**
+   * Filters db documents based on non document id field values if id use "$in" operator, use
+   * IDFilter, since partition key 'or' is not supported besides id, "$in" operator, use INFilter
+   */
+  public static class InFilter extends DBFilterBase {
+    private final Object arrayValue;
+    protected final InFilter.Operator operator;
+
+    @Override
+    JsonNode asJson(JsonNodeFactory nodeFactory) {
+      return DBFilterBase.getJsonNode(nodeFactory, arrayValue);
+    }
+
+    @Override
+    boolean canAddField() {
+      return false;
+    }
+
+    public enum Operator {
+      IN;
+    }
+
+    public InFilter(InFilter.Operator operator, String path, Object arrayValue) {
+      super(path);
+      this.arrayValue = arrayValue;
+      this.operator = operator;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) return true;
+      if (o == null || getClass() != o.getClass()) return false;
+      InFilter inFilter = (InFilter) o;
+      return operator == inFilter.operator && Objects.equals(arrayValue, inFilter.arrayValue);
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(arrayValue, operator);
+    }
+
+    private static final String DATA_CONTAINS = "array_contains";
```

Comment:
If this refers to name of DB column to use, I think we have a container class, `DocumentConstants` where we have added some column/field names. So perhaps add it there, instead of class private constant.

Reply:
solved